OrbitGl/TrackManager.h

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <stdint.h>  // for int32_t, uint64_t, uint32_t
+#include <stdlib.h>  // for abs
+
+#include <map>     // for map
+#include <memory>  // for shared_ptr
+#include <mutex>   // for recursive_mutex
+#include <string>  // for string
+#include <unordered_map>
+#include <vector>  // for vector
+
+#include "AsyncTrack.h"
+#include "FrameTrack.h"
+#include "GpuTrack.h"
+#include "GraphTrack.h"
+#include "PickingManager.h"  // for PickingMode
+#include "SchedulerTrack.h"
+#include "StringManager.h"  // for StringManager
+#include "ThreadTrack.h"
+#include "Timer.h"            // for Timer
+#include "Track.h"            // for Track
+#include "capture_data.pb.h"  // for FunctionInfo
+
+class TimeGraph;
+
+class TrackManager {
+ public:
+  explicit TrackManager(TimeGraph* time_graph);
+
+  void Clear();
+
+  [[nodiscard]] std::vector<Track*> GetTracks() const;
+  [[nodiscard]] std::vector<Track*> GetFilteredTracks() const { return sorted_filtered_tracks_; }
+  [[nodiscard]] std::vector<ThreadTrack*> GetThreadTracks() const;
+  [[nodiscard]] std::vector<FrameTrack*> GetFrameTracks() const;
+
+  [[nodiscard]] ThreadTrack* GetTracepointsSystemWideTrack() const {
+    return tracepoints_system_wide_track_;
+  }
+
+  void SetStringManager(std::shared_ptr<StringManager> str_manager);
+  [[nodiscard]] StringManager* GetStringManager() const { return string_manager_.get(); }
+
+  void SortTracks();
+  void SetFilter(const std::string& filter);
+  void UpdateFilteredTrackList();
```

Comment:
It was there before.

We update the Filtered (or Visible) track list when we change the filter and when we Sort Tracks (because we get the filtered from the sorted).

But maybe we can Sort directly both the visible and non-visible Tracks. I'm not seeing a priority here, let me know if I am wrong.

Reply:
I'm aware I'm blaming you for the bad choices of others (potentially even my own choices) :D
It's fine to have that method if it's used in two different places, but from what you describe it should be private then.

Reply:
UpdateFilteredTrackList could be inside SetFilter and also SortTracks. I was thinking when to make it, but better to make in this PR.

Reply:
I think you're missing the main point: I'm mostly arguing you should just move this to the `private` section since it's never called from outside as far as I can see. I'm fine with the method existing, but it should not be public.

Reply:
My bad!!! I was 100% sure that we had a reference to this method in TimeGraph and couldn't be private.